cmd/kpn/main.go

```diff
@@ -63,42 +55,28 @@ func init() {
 		nodecmd.AccountCommand,
 
 		// See utils/nodecmd/consolecmd.go:
-		nodecmd.GetConsoleCommand(nodeFlags, rpcFlags),
+		nodecmd.GetConsoleCommand(nodecmd.KpnNodeFlags(), nodecmd.CommonRPCFlags),
 		nodecmd.AttachCommand,
 
 		// See utils/nodecmd/versioncmd.go:
 		nodecmd.VersionCommand,
 
 		// See utils/nodecmd/dumpconfigcmd.go:
-		nodecmd.GetDumpConfigCommand(nodeFlags, rpcFlags),
+		nodecmd.GetDumpConfigCommand(nodecmd.KpnNodeFlags(), nodecmd.CommonRPCFlags),
 
 		// See utils/nodecmd/db_migration.go:
 		nodecmd.MigrationCommand,
 	}
 	sort.Sort(cli.CommandsByName(app.Commands))
 
-	app.Flags = append(app.Flags, nodeFlags...)
-	app.Flags = append(app.Flags, rpcFlags...)
-	app.Flags = append(app.Flags, nodecmd.ConsoleFlags...)
-	app.Flags = append(app.Flags, debug.Flags...)
-	app.Flags = append(app.Flags, nodecmd.DBMigrationFlags...)
+	app.Flags = nodecmd.KpnAppFlags()
 
 	cli.AppHelpTemplate = utils.GlobalAppHelpTemplate
 	cli.HelpPrinter = utils.NewHelpPrinter(utils.CategorizeFlags(app.Flags))
 
 	app.CommandNotFound = nodecmd.CommandNotExist
 
-	app.Before = func(ctx *cli.Context) error {
-		runtime.GOMAXPROCS(runtime.NumCPU())
-		logDir := (&node.Config{DataDir: utils.MakeDataDir(ctx)}).ResolvePath("logs")
-		debug.CreateLogDir(logDir)
-		if err := debug.Setup(ctx); err != nil {
-			return err
-		}
-		metricutils.StartMetricCollectionAndExport(ctx)
-		utils.SetupNetwork(ctx)
-		return nil
-	}
+	app.Before = nodecmd.BeforeRunNode
```

Comment:
As I see the implementation of any other cmds, this code looks different a little bit.
Is there any reason not to execute `nodecmd.CheckCommand` here like the below?
```golang
app.Before = func(ctx *cli.Context) error {
		if err := nodecmd.CheckCommands(ctx); err != nil {
			return err
		}
		return nodecmd.BeforeRunNode(ctx)
	}
```

Reply:
You are right.
- I have seen that each node has difference codes for app.Before
- And I analyzed it so that would like to modify it to like following in every nodes.
```go
app.Before = nodecmd.BeforeRunNode(ctx)
```
- But I couldn't assure 100% if It has no side-effect when operating the CN, PN, EN and tried to keep it same logic.
- So I hope we can discuss it together the next time we sort out unused flags.

Reply:
@kjhman21 @aidan-kwon Could you comment on this?

Reply:
I think this logic could be the same with other binaries.  It seems that this logic is missed during the refactoring

Reply:
You can refer old repository https://github.com/Krustuniverse-Klaytn-Group/klaytn/pull/3300/files

Reply:
Okay, thanks for the refference. I added the checkCommand to the BeforeRunNode in common